File: TileMapCreator/tilemap_scene.py
# tilemap_scene.py
from PySide6.QtWidgets import QGraphicsScene, QGraphicsPixmapItem
from PySide6.QtGui import QPixmap
from PySide6.QtCore import Qt, QPointF

class TileMapScene(QGraphicsScene):
    def __init__(self, tile_size=32):
        super().__init__()
        self.tile_size = tile_size
        self.active_tile = None  # Tuile sélectionnée
        self.setSceneRect(0, 0, 640, 480)
        self.setBackgroundBrush(Qt.lightGray)

    def set_active_tile(self, pixmap: QPixmap):
        self.active_tile = pixmap

    def mousePressEvent(self, event):
        if event.button() == Qt.LeftButton and self.active_tile:
            pos = event.scenePos()
            x = int(pos.x() // self.tile_size) * self.tile_size
            y = int(pos.y() // self.tile_size) * self.tile_size

            # Supprimer une tuile existante à cet emplacement
            for item in self.items(QPointF(x + 1, y + 1)):
                if isinstance(item, QGraphicsPixmapItem):
                    self.removeItem(item)

            # Placer la nouvelle tuile
            item = QGraphicsPixmapItem(self.active_tile)
            item.setPos(x, y)
            self.addItem(item)

File: TileMapCreator/CppRewrite/Sources/TileMapScene.cpp
#include "TileMapScene.h"
#include <QGraphicsItem>
#include <QGraphicsScene>
#include <QGraphicsSceneMouseEvent>
#include <QMouseEvent>
#include <qgraphicsitem.h>
#include <qgraphicsscene.h>
#include <qnamespace.h>
#include <qpixmap.h>
#include <qpoint.h>
#include <spdlog/spdlog.h>
#include <typeinfo>

TileMapScene::TileMapScene(int tile_size)
    : tile_size(tile_size), active_tile(QPixmap()) {
    setSceneRect(0, 0, 640, 480);
    setBackgroundBrush(Qt::lightGray);
}

void TileMapScene::set_active_tile(QPixmap pixmap) {
    spdlog::debug("Setting active tile...");
    active_tile = pixmap;
    spdlog::debug("Done");
}

void TileMapScene::set_tile_size(int t) { tile_size = t; }

void TileMapScene::mousePressEvent(QGraphicsSceneMouseEvent *event) {

    if (event->button() == Qt::LeftButton) {
        spdlog::debug("Click sur la scene des pixmap avec bouton gauche");
        QPointF pos = event->scenePos();
        int x = ((int)pos.x() / tile_size) * tile_size;
        int y = ((int)pos.y() / tile_size) * tile_size;

        int i = 0;
        while (i < items(QPointF(x + 1, y + 1)).size()) {
            if (typeid(items(QPointF(x + 1, y + 1))[i]) ==
                typeid(QGraphicsPixmapItem)) {
                items(QPointF(x + 1, y + 1)).removeAt(i);
            } else {
                i++;
            }
        }

        QGraphicsPixmapItem *item = new QGraphicsPixmapItem(active_tile);
        item->setPos(x, y);
        addItem(item);
    } else {
        spdlog::debug("Click sur la scene des pixmap");
    }
    QGraphicsScene::mousePressEvent(event);
}
